contrib/mesos/pkg/executor/executor_test.go

```diff
@@ -543,7 +543,7 @@ func NewMockPodsListWatch(initialPodList api.PodList) *MockPodsListWatch {
 		list:        initialPodList,
 	}
 	lw.ListWatch = cache.ListWatch{
-		WatchFunc: func(resourceVersion string) (watch.Interface, error) {
+		WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
```

Comment:
For better or worse (I think worse), in other contexts we made nil selector
mean nothing and nil-but-empty selector mean everything.

On Mon, Oct 26, 2015 at 9:14 PM, Clayton Coleman <notifications@github.com>
wrote:

> In contrib/mesos/pkg/executor/executor_test.go
> <https://github.com/kubernetes/kubernetes/pull/16266#discussion_r43083036>
> :
>
> > @@ -543,7 +543,7 @@ func NewMockPodsListWatch(initialPodList api.PodList) *MockPodsListWatch {
> >  		list:        initialPodList,
> >  	}
> >  	lw.ListWatch = cache.ListWatch{
> > -		WatchFunc: func(resourceVersion string) (watch.Interface, error) {
> > +		WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
>
> Yeah, I struggle with this as well. With Delete, we said backwards
> compatibility meant that "nil" was "default" and anything else was "user
> intent". I think with list, the implicit list is "all options defaulting",
> although for field and label selectors nil doesn't quite mean "everything"
> (although I guess it should).
> … <#150a7812cd53e24f_>
> On Tue, Oct 27, 2015 at 12:10 AM, Daniel Smith <notifications@github.com>
> wrote: In contrib/mesos/pkg/executor/executor_test.go <#16266 (comment)
> <https://github.com/kubernetes/kubernetes/pull/16266#discussion_r43082785>>
> : > @@ -543,7 +543,7 @@ func NewMockPodsListWatch(initialPodList
> api.PodList) *MockPodsListWatch { > list: initialPodList, > } >
> lw.ListWatch = cache.ListWatch{ > - WatchFunc: func(resourceVersion string)
> (watch.Interface, error) { > + WatchFunc: func(options api.ListOptions)
> (watch.Interface, error) { I feel like a pointer would be more natural, but
> I can't think of a concrete reason to ask you to switch... — Reply to this
> email directly or view it on GitHub <
> https://github.com/kubernetes/kubernetes/pull/16266/files#r43082785>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16266/files#r43083036>.
>